src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java

```diff
@@ -78,6 +81,12 @@
 
 	private final ListenerMode mode;
 
+	private final AtomicBoolean pausePending = new AtomicBoolean();
+
+	private final AtomicBoolean resumePending = new AtomicBoolean();
+
+	private final AtomicBoolean paused = new AtomicBoolean();
```

Comment:
> Delegate Pausable methods to listener container 

Therefore we don't need all of these properties anymore

Reply:
Doh